src/box/tuple.h

```diff
@@ -1125,19 +1182,11 @@ tuple_ref_slow(struct tuple *tuple);
 static inline void
 tuple_ref(struct tuple *tuple)
 {
-	if (unlikely(tuple->refs >= TUPLE_REF_MAX))
-		tuple_ref_slow(tuple);
-	else
-		tuple->refs++;
+	if (unlikely(tuple->local_refs >= TUPLE_REF_MAX))
+		tuple_upload_refs(tuple);
```

Comment:
Shouldn't be here return? I mean tuple->local_refs are incremented unconditionally..

Reply:
Nevermind, I got the logic.